app/models/titles_list.rb

```diff
@@ -0,0 +1,94 @@
+# Encapsulates the title logic for a Media, making it much easier to work with.
+class TitlesList
+  # @param titles [{Symbol, String => String}] the titles in various locales
+  # @param canonical_title_key [Symbol, String] the key of the canonical title in the titles hash
+  # @param alternatives [<String>] the list of alternative titles
+  # @param original_languages [<String>] the list of languages the media was originally released in
+  # @param original_countries [<String>] the list of countries the media was originally released in
+  def initialize(
+    titles:,
+    canonical_locale:,
+    alternatives: [],
+    original_languages: [],
+    original_countries: []
+  )
+    @titles = titles.symbolize_keys
+    @canonical_locale = canonical_locale.to_sym
+    @alternatives = alternatives
+    @original_languages = original_languages
+    @original_countries = original_countries
+  end
+
+  # @return [{Symbol => String}] the titles keyed by locale
+  def localized
+    @titles
+  end
+
+  # @return [<String>] the list of alternative titles
+  attr_reader :alternatives
+
+  # @return [Symbol] the key of the canonical title in the localized titles hash
+  attr_reader :canonical_locale
+
+  # @return [Symbol, nil] the key of the romanized/romaji title in the localized titles hash
+  def romanized_locale
+    # HACK: this should use originalLanguages, but we can't easily use that until we switch to using
+    # the en-t-ja style keys.  For now, this is the same bodge from the frontend.
+    %i[en_cn en_kr en_jp].find { |key| @titles.key?(key) }
+    if @titles.key?(:en_cn) then :en_cn
+    elsif @titles.key?(:en_kr) then :en_kr
+    elsif @titles.key?(:en_jp) then :en_jp
+    end
+  end
+
+  # @return [Symbol, nil] the key of the original title of the media in the localized titles hash
+  def original_locale
+    # TODO: we should aim to resolve all cases where this is nil or uses the fallback logic
+    preferred_key = "#{@original_languages.first}_#{@original_countries.first}".to_sym
```

Comment:
most likely worth just extracting this out into a private method.

Reply:
It's used once and it's the only piece of logic here which *isn't* about dealing with missing data (and thus will be resolved once we fill the data in)

I don't think it's necessary to extract